Condition with stateful parameter

mraunak opened the pull request:
Stateful_parameter is added to abc_condition examples

pgleeson commented:
@mraunak 2 small things:
- Can you simplify the methods in the nodes to just count the number of times the nodes get run, so if the run sequence is AAABB the final state value of A is 3 and B=2? Also, can you regenerate the images for the conditional example
- There are some documentation changes in here, incl changing the main Specification.md to rst, which is fine, but it would be good to either put these changes elsewhere until there is a script to convert the generated md to rst, or add the steps to do that automatically (e.g. in https://github.com/ModECI/MDF/blob/main/docs/generate.py)

mraunak commented:
The changes made:
1. Added a parameter at each node to just count the number of times the node ran 
2. Regenerated the images for the conditional example using Graphviz
3. Removed docs changes from this pull request

pgleeson commented:
Looks good @mraunak!
